src/net.py

```diff
@@ -588,10 +606,16 @@ def train(self, epochs, train_x, train_y, val_x, val_y,
                         )
                         # print ('to {}'.format(self.learning_rate))
                     self.minibatch_iteration += 1
+
                 train_error, train_accuracy = self.validator(train_x, train_y)
                 validation_error, validation_accuracy = self.validator(val_x,
                                                                        val_y)
 
+                if self.stopping_rule == 'best_validation' and validation_error < best_error:
+                    best_state = self.__getstate__()
```

Comment:
I think to save more memory, it might be a better idea to just save the parameters using a similar command to ```net_parameters = np.array(lasagne.layers.get_all_param_values(self.layers,**{self.name: True}))``` found in the __getstate__ function. using just __getstate__ will copy over other parts of the class that are not necessary.

Reply:
I figured this was slightly safer than copying the weights and performance data over (since we might add stuff and forget to add it to the copier), but it's your call.

Reply:
Im just worried about the additional memory overhead that this would introduce after training for a while.